napari/utils/notebook_display.py

```diff
@@ -85,9 +86,20 @@ def _clean_alt_text(self, alt_text):
             # cleaner won't recognize escaped script tags, so always unescape
             # to be safe
             alt_text = html.unescape(str(alt_text))
-            doc = document_fromstring(alt_text)
-            alt_text = Cleaner().clean_html(doc).text_content()
-        return alt_text or None
+            cleaner = Cleaner()
+            try:
+                doc = document_fromstring(alt_text)
+            except ParserError:
+                alt_text = ""
+                warn(
+                    'The provided alt text does not constitute valid html, so it was discarded.',
+                    stacklevel=3,
+                )
+            alt_text = cleaner.clean_html(doc).text_content()
```

Comment:
@brisvag Currently this is an UnboundLocalError, because `doc` fails to be assigned in the try clause. Reproducer:

```python
import napari
from napari.utils.notebook_display import NotebookScreenshot

viewer = napari.Viewer()
nbscr = NotebookScreenshot(viewer, alt_text=" ")
```

So this needs to be fixed and then a test added for good measure. 😂

Reply:
oops, things got entangled in refactoring. Should work now!